include/na/azac/VMPlacer.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "ir/Definitions.hpp"
+#include "na/azac/Architecture.hpp"
+#include "na/azac/Types.hpp"
+
+#include <cstddef>
+#include <functional>
+#include <nlohmann/json_fwd.hpp>
+#include <optional>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+namespace na::azac {
+/// class to find a qubit layout based on vertex matching of a weighted
+/// bipartite graph
+class VMPlacer {
+  friend class VMPlacerTest_MinimumWeightFullBipartiteMatching1_Test;
+  friend class VMPlacerTest_MinimumWeightFullBipartiteMatching2_Test;
+  friend class VMPlacerTest_MinimumWeightFullBipartiteMatchingExceptions_Test;
+  friend class VMPlacerTest_MinimumWeightFullBipartiteMatchingEmpty_Test;
+
+  std::reference_wrapper<const Architecture> architecture_;
+  /// If true, during the initial placement the atoms are placed starting in the
+  /// last row instead of the first row in the first SLM
+  bool reverseInitialPlacement_ = false;
+
+  /// this flag indicates whether the  placement should use a window when
+  /// selecting potential free sites
+  bool useWindow_ = true;
+  size_t windowSize_ = 0;
+
+  /// this flag indicates whether the placement between gates is dynamic, i.e.,
+  /// if this flag is false, the initial placement is used after all gates
+  bool dynamicPlacement_ = true;
+
+  // todo: Why is that?
+  constexpr static double costAtomTransfer_ = 0.9999;
+
+public:
+  /// Create a VMPlacer based on the given architecture and configuration
+  VMPlacer(const Architecture& architecture, const nlohmann::json& config);
+  /// generate qubit placement based on minimum weight matching
+  [[nodiscard]] auto
+  place(size_t nQubits,
+        const std::vector<TwoQubitGateLayer>& twoQubitGateLayers,
+        const std::vector<std::unordered_set<qc::Qubit>>& reuseQubits)
+      -> std::vector<Placement>;
```

Comment:
Given how this is the main method provided by this class, making this the defacto public interface, this is missing an appropriately elaborate docstring.